pkgs/development/python-modules/wacz/default.nix

```diff
@@ -0,0 +1,71 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  boilerpy3,
+  cdxj-indexer,
+  frictionless,
+  jsonlines,
+  pytest-cov,
+  pyyaml,
+  shortuuid,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "wacz";
+  version = "0.5.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "webrecorder";
+    repo = "py-wacz";
+    tag = "v${version}";
+    hash = "sha256-bGY6G7qBAN1Vu+pTNqRG0xh34sR62pMhQFHFGlJaTPQ=";
+  };
+
+  postPatch = ''
+    substituteInPlace setup.py \
+      --replace "pytest-runner" ""
+  '';
+
+  # Upstream appears to attempt to pin transitive dependencies (typer, click)
+  # to work around package conflicts.
+  dependencies = [
+    boilerpy3
+    cdxj-indexer
+    frictionless
+    pytest-cov
+    pyyaml
+    shortuuid
+  ];
+
+  optional-dependencies = {
+    # signing = [
+    #   authsign # not packaged
+    # ];
+  };
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    jsonlines # Not even actually used...
```

Comment:
What do you mean here exactly? Does upstream use `import jsonlines` but they could in principal delete this `import`?

Reply:
Right, the `import jsonlines` is the only place where it's used, and from the git history it appears it's _never_ been actually used.

Reply:
> Right, the `import jsonlines` is the only place where it's used, and from the git history it appears it's _never_ been actually used.

Then we should report upstream it is not required - I'm sure they'll appreciate our feedback. In general we don't want to add dependency links between packages unless it is absolutely needed.

Reply:
I opened https://github.com/webrecorder/py-wacz/pull/47 to clean this up. `jsonlines` is used transitively through `frictionless` if it's in the PYTHONPATH and `wacz` should depend simply on `frictionless[json]`.